src/index.ts

```diff
@@ -26,53 +26,40 @@
 
 'use strict'
 
-
-// import { DefaultEventLogger } from './DefaultEventLogger'
-import { EventLoggingServiceServer, EVENT_RECEIVED } from "./transport/EventLoggingServiceServer";
-import { EventPostProcessor } from './EventPostProcessor';
-import { EventPreProcessor } from './EventPreProcessor';
 import { Tracer } from './Tracer'
 
-import {   EventMessage,
-  EventType,
+import { DefaultLoggerRecorder, DefaultSidecarRecorder } from './Recorder'
+
+import {
+  EventMessage,
   LogEventTypeAction,
   AuditEventTypeAction,
   TraceEventTypeAction,
   LogEventAction,
   AuditEventAction,
   TraceEventAction,
   EventStatusType,
-  IMessageMetadata,
   EventMetadata,
-  EventTraceMetadata,
   LogResponseStatus,
-  LogResponse
+  LogResponse,
+  TypeMessageMetadata
  } from "./model/EventMessage"
 
-
-// let logger = new DefaultEventLogger()
-
 // Re-export definitions and components on the public API
 export {
-  // logger,
-  // DefaultEventLogger,
-  EventLoggingServiceServer,
-  EVENT_RECEIVED,
-  EventPostProcessor,
-  EventPreProcessor,
-  EventMessage,
-  EventType,
   LogEventTypeAction,
   AuditEventTypeAction,
   TraceEventTypeAction,
   LogEventAction,
   AuditEventAction,
   TraceEventAction,
   EventStatusType,
-  IMessageMetadata,
   EventMetadata,
-  EventTraceMetadata,
+  EventMessage,
   LogResponseStatus,
   LogResponse,
-  Tracer
+  TypeMessageMetadata,
+  Tracer,
+  DefaultLoggerRecorder,
+  DefaultSidecarRecorder
```

Comment:
Please export the Span, and also interfaces as required to enable someone to create new Tracers, Spans, etc, or alternatively extend the existing implementations.